### ISSUE TITLE
how to stop sorting in select2 after fetching the data in select box or refreshing the page 

### ISSUE THREAD

**Issue by TayyabAslam123** [select2/select2#6172]
Currrently I am doing this:  

 <select class="form-control profileName-multiple" name="profileName[]" multiple>
                <?php
                    foreach($profileList as $pk=>$pv){
                        echo '<optgroup label="'.$pk.'">';
                        foreach($pv as $profile){
                            $selected = '';
                            if(in_array($profile->id, $saveProfileList)){
                                $selected = ' selected ';
                            }
                            echo '<option '.$selected.' value="'.$profile->id.'">'.$profile->name.'</option>';
                        }
                        echo '</optgroup>';
                    }
                ?>
                </select>
                
                
     $('.profileName-multiple').select2().on("select2:select", function (evt) {
            var element = evt.params.data.element;
            var $element = $(element);
            
            $element.detach();
            $(this).append($element);
            $(this).trigger("change");
        })

**Comment by kevin-brown** [comment]
It's not possible to stop this "sorting" without fixing the underlying problem: you're moving the location of the option within the `<select>` box by appending the option. Select2 will always reflect the order of the underlying options, which happens to also be the same order the values are sent to the server (which is important).
